editor/src/components/inspector/sections/component-section/component-section.tsx

```diff
@@ -1164,54 +1184,81 @@ export class ComponentSection extends React.Component<
 
 interface VariableOption {
   variableName: string
+  displayName: string
   definedElsewhere: string | null
   value: string
+  depth?: number
 }
 
 function valuesFromObject(
   name: string,
   objectName: string,
   value: object | null,
+  depth: number = 0,
+  displayName: string = name,
 ): Array<VariableOption> {
   if (value == null) {
-    return [{ variableName: name, definedElsewhere: null, value: `null` }]
+    return [{ displayName, variableName: name, definedElsewhere: null, value: `null` }]
   }
 
   const patchDefinedElsewhereInfo = (variable: VariableOption) => ({
     variableName: variable.variableName,
     value: variable.value,
     definedElsewhere: objectName,
+    displayName: variable.displayName,
+    depth: variable.depth,
   })
 
   if (Array.isArray(value)) {
-    const values = value.flatMap((v, idx) =>
-      valuesFromVariable(`${name}[${idx}]`, v).map((variable) =>
-        patchDefinedElsewhereInfo(variable),
+    return [
+      patchDefinedElsewhereInfo({
+        displayName,
+        variableName: name,
+        definedElsewhere: objectName,
+        value: `[ ]`,
+        depth,
+      }),
+    ].concat(
+      value.flatMap((v, idx) =>
+        valuesFromVariable(`${name}[${idx}]`, v, depth + 1, `${displayName}[${idx}]`).map(
+          (variable) => patchDefinedElsewhereInfo(variable),
+        ),
       ),
     )
-
-    return [{ variableName: name, definedElsewhere: name, value: '[array]' }, ...values]
   }
 
-  const values = Object.entries(value).flatMap(([key, field]) =>
-    valuesFromVariable(`${name}['${key}']`, field).map((variable) =>
-      patchDefinedElsewhereInfo(variable),
+  return [
+    patchDefinedElsewhereInfo({
+      displayName,
```

Comment:
nitpick, but we don't like to mix the verbose object literal and the shorthand version. we actually prefer the verbose `displayName: displayName` over the shorthand

Reply:
good point, fixed!